platforms/gemstone/projects/ston/src/STON-GemStoneBase/STONReader.extension.st

```diff
@@ -10,7 +10,10 @@ STONReader >> lookupClass: name [
 					ifNotNil: [:cls | ^cls ] ])
 						ifNil: [ classes at: name ifAbsentPut: [ (ClassOrganizer new allSubclassesOf: Object)
 								detect: [ :cls | cls stonName == name ]
-								ifNone: [ (Rowan platform serviceClassFor: name) ifNil:[self error: 'Cannot resolve class named ' , name printString ]] ] ] ]
+								ifNone: [
+									(((AllUsers userWithId: 'SystemUser') objectNamed: 'Rowan') 
+										platform serviceClassFor: name)
```

Comment:
At some point, I might consider factoring the code that returns 'SystemUser' out into it's own method to avoid having a constant buried ... but that's a minor point. 

Question - Is there ever a case where a user would not have access to ``(AllUsers userWithId: 'SystemUser')``?

Reply:
AFAIK all users have read  permissions for AllUsers ... if the user does not have permissions to see AllUsers, they probably aren't allowed to read code either and shouldn't be using Oscar:)

I don't like this SystemUser stuff either, but rather than "hide it", I would like to make it pluggable in some way, since this is not necessarily an uncommon problem ... hmmm, perhaps a better approach would be to add a lookup resolution block to the fromString: api ... that way each individual read invocation could customize the lookup namespace ... I think I like this approach ... we can discuss when I get into office ...